src/features/view-nft/NFTDetailsCard/ui/Options/helpers.tsx

```diff
@@ -0,0 +1,151 @@
+import { ReactNode } from 'react';
+
+import { OptionLabel } from '../../../../ui';
+import { ViewBidsModal } from '../../../../view-bids';
+import { SetBuyNowModal } from '../../../../set-buy-now';
+import { CreateTokenModal } from '../../../../create-token';
+import { DomainSettingsModal } from '../../../../domain-settings';
+import { TransferOwnershipModal } from '../../../../transfer-ownership';
+import {
+	IconCube1,
+	IconCurrencyDollar,
+	IconDatabase2,
+	IconImage1,
+	IconSend3,
+	IconTag1,
+	IconVideoRecorder,
+} from '@zero-tech/zui/components/Icons';
+
+export const enum DropdownOptionType {
+	CREATE_TOKEN = 'create-token',
+	DOMAIN_SETTINGS = 'domain-settings',
+	SET_BUY_NOW = 'set-buy-now',
+	TRANSFER = 'transfer',
+	VIEW_BIDS = 'view-bids',
+}
+
+export type DropdownOption =
+	| 'create-token'
+	| 'domain-settings'
+	| 'set-buy-now'
+	| 'transfer'
+	| 'view-bids';
+
+/**
+ * Gets the download options and filters by condition
+ * @param downloadAsset input - uses useDownloadAsset hook return
+ * @param imageUrl input - metadata image url
+ * @param imageUrl2 input - metadata image_2 url
+ * @param imageUrl3 input - metadata image_3 url
+ * @param animationUrl input - metadata animation_url
+ * @returns
+ */
+export const getDownloadOptions = (
+	downloadAsset: (assetUrl: string) => Promise<void>,
+	imageUrl?: string,
+	imageUrl2?: string,
+	imageUrl3?: string,
+	animationUrl?: string,
+) => {
+	const args: GetUrlArgs = {
+		imageUrl,
+		imageUrl2,
+		imageUrl3,
+		animationUrl,
+	};
+
+	return downloadOptionConfig
+		.filter(({ getUrl }) => getUrl(args))
+		.map(({ id, label, icon, getUrl }) => ({
+			id,
+			label: <OptionLabel label={label} icon={icon} />,
+			onSelect: () => downloadAsset(getUrl(args)),
+		}));
+};
```

Comment:
nit: potentially might be overkill - I'd say things would be tidier/more efficient if we just hardcoded the rendering of these, i.e.

```tsx
{imageUrl && <OptionLabel label={label} icon={icon} onSelect={() => downloadAsset(getUrl(imageUrl)} />}
{imageUrl2 && <OptionLabel label={label} icon={icon} onSelect={() => downloadAsset(getUrl(imageUrl2)} />}
{imageUrl3 && <OptionLabel label={label} icon={icon} onSelect={() => downloadAsset(getUrl(imageUrl3)} />}
```

I think these functions are good when they reduce complexity/make it easier to add new items, but I think it's also easy to lose sight a little bit and end up with something more complicated than hardcoding.

Reply:
Ok, I wasn't following this one correct - I see what's happening here, you're setting up the objects to send to `DropdownMenu` rather than conditionally rendering. This makes more sense - ignore the above feedback, but [this feedback](https://github.com/zer0-os/zApp-NFTs/pull/89#discussion_r1171961761) should apply here